core/src/main/java/io/grpc/LoadBalancer2.java

```diff
@@ -317,6 +322,77 @@ public String toString() {
   }
 
   /**
+   * Provides arguments for a {@link SubchannelPicker#pickSubchannel(PickSubchannelArgs)}.
+   */
+  public abstract static class PickSubchannelArgs {
```

Comment:
I agree with you it's a bit inconvenient for test writers. Because it appears to me both approaches have about the same amount of problems, I would opt for a simpler API, which is the abstract class.